src/windowmanagement/TaskBarMenuChoices.tsx

```diff
@@ -27,37 +32,51 @@ export function TaskBarMenuChoices({
       arrowPosition={'center'}
     >
       <Menu.Target>
-        <Menu.Item>Task Bar</Menu.Item>
+        <Menu.Item
+          leftSection={<TaskBarIcon />}
+          rightSection={<ChevronRightIcon size={IconSize.sm} />}
+        >
+          Task Bar
+        </Menu.Item>
       </Menu.Target>
       <Menu.Dropdown>
         <Menu.Label>Toggle Task Bar Items</Menu.Label>
-        <Stack gap={'xs'}>
-          {menuItemsDB.map((item) => (
-            <Checkbox
-              key={item.componentID}
-              label={item.title}
-              defaultChecked={checkedMenuItems[item.componentID]}
-              onChange={(event) => {
-                const { checked } = event.currentTarget;
-                setVisibleMenuItems((prevstate) => {
-                  if (checked) {
-                    // This check is necessary because the callback is sometimes called
-                    // twice, meaning we would otherwise add the item multiple times
-                    if (prevstate.includes(item.componentID)) {
-                      return prevstate;
-                    }
-                    prevstate.push(item.componentID);
-                  } else {
-                    const index = prevstate.findIndex((id) => id === item.componentID);
-                    if (index >= 0) {
-                      prevstate.splice(index, 1);
-                    }
-                  }
-                  return [...prevstate];
-                });
-              }}
-            />
-          ))}
+        <Stack gap={0}>
+          {menuItemsDB.map((item) => {
+            return (
+              <Group className={'ToggleTaskBarItemParent'}>
+                {item.icon?.(IconSize.xs)}
+                <Checkbox
+                  className={'ToggleTaskBarItem'}
+                  key={item.componentID}
+                  label={item.title}
+                  labelPosition={'left'}
+                  defaultChecked={checkedMenuItems[item.componentID]}
+                  onChange={(event) => {
+                    const { checked } = event.currentTarget;
+                    setVisibleMenuItems((prevstate) => {
+                      if (checked) {
+                        // This check is necessary because the callback is sometimes called
+                        // twice, meaning we would otherwise add the item multiple times
+                        if (prevstate.includes(item.componentID)) {
+                          return prevstate;
+                        }
+                        prevstate.push(item.componentID);
```

Comment:
I don't think it is recommended to modify function arguments like this. Consider spreading the prevstate first before modifying it